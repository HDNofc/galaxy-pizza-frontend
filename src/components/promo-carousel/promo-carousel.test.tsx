import renderWithThemeProvider from 'test-utils/render-with-theme-provider';
import PromoCarousel from './promo-carousel';

const stubs = [
  {
    id: 'id-1',
    src: 'image-url-1',
    alt: 'alt-1',
  },
  {
    id: 'id-2',
    src: 'image-url-2',
    alt: 'alt-2',
  },
  {
    id: 'id-3',
    src: 'image-url-3',
    alt: 'alt-3',
  },
];

describe('<PromoCarousel>', () => {
  const defaultProps = {
    items: stubs,
  };

  it('отображается корректно с верным количество элементов', () => {
    const { container } = renderWithThemeProvider(<PromoCarousel {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
