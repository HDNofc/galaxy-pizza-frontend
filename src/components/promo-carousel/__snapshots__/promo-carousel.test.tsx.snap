// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromoCarousel> отображается корректно с верным количество элементов 1`] = `
<div
  class="PromoCarousel-sc-1l9778m-0 gfIHaU"
>
  <div
    class="swiper-container swiper-container-initialized swiper-container-horizontal"
  >
    <div
      class="swiper-wrapper"
      style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
    >
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="0"
      >
        <div
          class="SlideContainer-sc-1l9778m-1"
        >
          <img
            alt="alt-1"
            class="SlideImage-sc-1l9778m-2 jyXlEq"
            src="image-url-1"
          />
        </div>
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="1"
      >
        <div
          class="SlideContainer-sc-1l9778m-1"
        >
          <img
            alt="alt-2"
            class="SlideImage-sc-1l9778m-2 jyXlEq"
            src="image-url-2"
          />
        </div>
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="2"
      >
        <div
          class="SlideContainer-sc-1l9778m-1"
        >
          <img
            alt="alt-3"
            class="SlideImage-sc-1l9778m-2 jyXlEq"
            src="image-url-3"
          />
        </div>
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="0"
      >
        <div
          class="SlideContainer-sc-1l9778m-1"
        >
          <img
            alt="alt-1"
            class="SlideImage-sc-1l9778m-2 jyXlEq"
            src="image-url-1"
          />
        </div>
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="1"
      >
        <div
          class="SlideContainer-sc-1l9778m-1"
        >
          <img
            alt="alt-2"
            class="SlideImage-sc-1l9778m-2 jyXlEq"
            src="image-url-2"
          />
        </div>
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="2"
      >
        <div
          class="SlideContainer-sc-1l9778m-1"
        >
          <img
            alt="alt-3"
            class="SlideImage-sc-1l9778m-2 jyXlEq"
            src="image-url-3"
          />
        </div>
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="0"
      >
        <div
          class="SlideContainer-sc-1l9778m-1"
        >
          <img
            alt="alt-1"
            class="SlideImage-sc-1l9778m-2 jyXlEq"
            src="image-url-1"
          />
        </div>
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="1"
      >
        <div
          class="SlideContainer-sc-1l9778m-1"
        >
          <img
            alt="alt-2"
            class="SlideImage-sc-1l9778m-2 jyXlEq"
            src="image-url-2"
          />
        </div>
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="2"
      >
        <div
          class="SlideContainer-sc-1l9778m-1"
        >
          <img
            alt="alt-3"
            class="SlideImage-sc-1l9778m-2 jyXlEq"
            src="image-url-3"
          />
        </div>
      </div>
    </div>
    <ul />
  </div>
</div>
`;
